lucille-plugins/lucille-video/src/test/java/com/kmwllc/lucille/video/VideoFileHandlerTest.java

```diff
@@ -0,0 +1,215 @@
+      Iterator<Document> docs = strideHandler.processFile(fis, filePath);
+      while (docs.hasNext()) {
+        Document doc = docs.next();
+        Integer frameIndex = doc.getInt("frame_index");
+        assertNotNull(frameIndex);
+        strideIndices.add(frameIndex);
+      }
+    }
+
+    assertTrue(strideIndices.size() < defaultIndices.size());
+
+    int maxComparisons = Math.min(strideIndices.size(), defaultIndices.size() / 4);
+
+    for (int i = 0; i < maxComparisons; i++) {
+      int expectedIndex = defaultIndices.get(i * 4);
+      int actualIndex = strideIndices.get(i);
+      assertEquals(expectedIndex, actualIndex);
+    }
+  }
+
+  @Test
+  public void testConfigurableFieldNames() throws Exception {
+    // Use non-default field names for every configurable field
+    Config config = ConfigFactory.parseMap(Map.of(
+        "sourceField", "src_path",
+        "fileNameField", "fname",
+        "frameIndexField", "frame_idx",
+        "frameTimeMsField", "frame_ms",
+        "frameTimecodeField", "tc",
+        "imageWidthField", "w",
+        "imageHeightField", "h",
+        "frameImageField", "png"
+    ));
+    VideoFileHandler handler = new VideoFileHandler(config);
+
+    String filePath = "src/test/resources/VideoFileHandlerTest/sample.mp4";
+    File file = new File(filePath);
+    String leaf = FilenameUtils.getName(filePath);
+
+    try (FileInputStream fis = new FileInputStream(file)) {
+      Iterator<Document> docs = handler.processFile(fis, filePath);
+      Document first = docs.next();
+
+      assertEquals(filePath, first.getString("src_path"));
+      assertEquals(leaf, first.getString("fname"));
+
+      Integer frameIdx = first.getInt("frame_idx");
+      assertNotNull(frameIdx);
+
+      Long frameMs = first.getLong("frame_ms");
+      assertNotNull(frameMs);
+      assertTrue(frameMs >= 0L);
+
+      String timecode = first.getString("tc");
+      assertNotNull(timecode);
+
+      Integer width = first.getInt("w");
+      Integer height = first.getInt("h");
+      assertNotNull(width);
+      assertNotNull(height);
+      assertTrue(width > 0);
+      assertTrue(height > 0);
+
+      byte[] imageBytes = first.getBytes("png");
+      assertNotNull(imageBytes);
+      assertTrue(imageBytes.length > 0);
+    }
+  }
+
+  @Test
+  public void testFrameTimecodeFormatting() throws Exception {
+    Config config = ConfigFactory.empty();
+    VideoFileHandler handler = new VideoFileHandler(config);
+
+    String filePath = "src/test/resources/VideoFileHandlerTest/sample.mp4";
+    File file = new File(filePath);
+
+    try (FileInputStream fis = new FileInputStream(file)) {
+      Iterator<Document> docs = handler.processFile(fis, filePath);
+      Document first = docs.next();
+
+      Long frameTimeMs = first.getLong("frame_time_ms");
+      assertNotNull(frameTimeMs);
+
+      String timecode = first.getString("frame_timecode");
+      assertNotNull(timecode);
+
+      String expectedTimecode = "00:00:00.033";
```

Comment:
nitpicky but I would keep this test and add another one where you call `formatTimecode(long millis)` directly, with a variety of values so that the elements of the timecode come out to be something other than 00. You can make `formatTImecode` package-private (so you can access it in the test) just by removing the `private` modifier.

also for brevity in the testing you don't really need to declare `String expectedTimecode` if you are going to use it in an assertion immediately, just place the string literal right in the assertion